lib/AST/DiagnosticEngine.cpp

```diff
@@ -940,3 +941,19 @@ BufferIndirectlyCausingDiagnosticRAII::BufferIndirectlyCausingDiagnosticRAII(
   if (loc.isValid())
     Diags.setBufferIndirectlyCausingDiagnosticToInput(loc);
 }
+
+void DiagnosticEngine::onTentativeDiagnosticFlush(Diagnostic &diagnostic) {
+  for (auto curr = diagnostic.Args.begin(), last = diagnostic.Args.end();
```

Comment:
Why not a for-each loop using references?

Reply:
Hm, indeed why. I just remembered that std::replace if is implemented similarly but I can switch to use refs instead!

Reply:
Done!